Jenkinsfile

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env groovy
+
+def mainBranches() {
+    return BRANCH_NAME == "develop" || BRANCH_NAME.startsWith("release/");
+}
+
+// TODO: Use multiple choices
+run_linter = true
+rust_test = true
+helm_test = true
+run_tests = params.run_tests
+
+// Will skip steps for cases when we don't want to build
+if (currentBuild.getBuildCauses('jenkins.branch.BranchIndexingCause') && mainBranches()) {
+    print "INFO: Branch Indexing, skip tests and push the new images."
+    run_tests = false
+}
+
+// Only schedule regular builds on main branches, so we don't need to guard against it
+String cron_schedule = mainBranches() ? "0 2 * * *" : ""
+
+pipeline {
+  agent none
+  options {
+    timeout(time: 1, unit: 'HOURS')
+  }
+  parameters {
+    booleanParam(defaultValue: true, name: 'run_tests')
+  }
+  triggers {
+    cron(cron_schedule)
+  }
+
+  stages {
+    stage('init') {
+      agent { label 'nixos-mayastor' }
+      steps {
+        step([
+          $class: 'GitHubSetCommitStatusBuilder',
+          contextSource: [
+            $class: 'ManuallyEnteredCommitContextSource',
+            context: 'continuous-integration/jenkins/branch'
+          ],
+          statusMessage: [ content: 'Pipeline started' ]
+        ])
+      }
+    }
+    stage('linter') {
+      agent { label 'nixos-mayastor' }
+      when {
+        beforeAgent true
+        not {
+          anyOf {
+            branch 'master'
+            branch 'release/*'
+            expression { run_linter == false }
```

Comment:
we don't have a linter here, but I guess we could use a linter to lint values.yaml in the future?